### PR TITLE
React & Redux group project - Space Travelers' Hub 

### PR DESCRIPTION
**Pull request by alex1779**
[Project requirements]

- [Config & basic setup](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#config--basic-setup)
- Create React App.
- Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/react_router.md).
- Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.

- Create routes and view components: rockets, missions, my profile. Add dragons [only if your team has 3 members]).

- Use for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).

- Create directories for all Redux state slice files:rockets, missions.

- [Redux: Fetch data and update Redux store](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store)

- Upon first render fetch data from the SpaceX API endpoints:

        Rockets: https://api.spacexdata.com/v4/rockets
        Missions: https://api.spacexdata.com/v3/missions

Once the data are fetched, dispatch an action to store the selected data in Redux store:

    Rockets:
        id
        rocket_name
        description
        flickr_images

    Missions:
        mission_id
        mission_name
        description

- [Render UI:lists](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#render-uilists)

- [Redux: Write actions and reducers for booking rockets/dragons and joining missions](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-write-actions-and-reducers-for-booking-rocketsdragons-and-joining-missions)

- [Redux: Write actions and reducers for canceling rockets/dragons and leaving missions](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-write-actions-and-reducers-for-canceling-rocketsdragons-and-leaving-missions)

- [Render UI: conditional components rendering](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#render-ui-conditional-components-rendering)

- [Render UI: My Profile section](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#render-ui-my-profile-section)

- [Technical set up**](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#technical-set-up)

    Set up the repository on GitHub and use Gitflow.
    Set up React and Redux.
    Set up testing libraries(React Testing Library and Jest).
    Do not preserve user data.

### CODE REVIEW

File: src/redux/missions/missionsSlice.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    try {
+      const response = await fetch('https://api.spacexdata.com/v3/missions');
+      const missions = await response.json();
+      return missions;
+    } catch (error) {
+      return rejectWithValue(error.message);
+    }
+  },
+);
+
```
Comment (julie-ify) [inline src/redux/missions/missionsSlice.js]: - [ ] Kindly filter the mission data you fetched from the API based on the requirement [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store)

File: src/redux/rockets/rocketsSlice.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  async (_, { rejectWithValue }) => {
+    try {
+      const response = await fetch('https://api.spacexdata.com/v4/rockets');
+      const rockets = await response.json();
+      return rockets;
+    } catch (error) {
+      return rejectWithValue(error.message);
+    }
+  },
+);
```
Comment (julie-ify) [inline src/redux/rockets/rocketsSlice.js]: - [ ] We don't need to save all the data from the API. Kindly filter the data based on [this requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store).

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @alex1779 and @joramot,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/missions/missionsSlice.js (hunk 1 above)
Comment (alex1779) [reply src/redux/missions/missionsSlice.js]: Hi @julie-ify , Could you explain to me what you mean by applying a filter? I don't see in the requirements that that is mentioned.

Comment (julie-ify) [reply src/redux/missions/missionsSlice.js]: Hi @alex1779 sure

Comment (julie-ify) [reply src/redux/missions/missionsSlice.js]: Look at the missions data for example. All the key-value pairs from the API are saved to the store. But from the sub-section of the requirement I shared with you only `mission_id`, `mission_name`, and `description` should be saved.

image

File: src/redux/rockets/rocketsSlice.js (hunk 2 above)
Comment (julie-ify) [reply src/redux/rockets/rocketsSlice.js]: For the rocket data, only the `id`, `rocket_name`, `description`, and `flickr_images` should be saved.

image

File: src/redux/missions/missionsSlice.js (hunk 1 above)
Comment (julie-ify) [reply src/redux/missions/missionsSlice.js]: PS. If you wish to have see the store in the browser, like I have done you need to configure the `redux-logger` in the store file.

  ```
  import { configureStore } from '@reduxjs/toolkit';
  import logger from 'redux-logger';
  import rocketsReducer from './rockets/rocketsSlice';
  import missionsReducer from './missions/missionsSlice';
  
  const store = configureStore({
    reducer: {
      missions: missionsReducer,
      rockets: rocketsReducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  });
  ```

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @alex1779, @joramot, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 


## Optional suggestions

- [ ] **OPTIONAL** As the previous reviewer suggested, when you are using third-party APIS is often a good practice to manipulate the data you receive to only the one you're going to use, for example, these APIS returns a lot of data and you're just using 3 or 4 properties. This could keep your store better organized! 🤓 🔧 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**